cmd/fossa/update.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/blang/semver"
+	logging "github.com/op/go-logging"
+	"github.com/rhysd/go-github-selfupdate/selfupdate"
+	"github.com/urfave/cli"
+)
+
+var updateLogger = logging.MustGetLogger("update")
+var updateEndpoint = "fossas/fossa-cli"
+
+func getSemver() string {
+	if version[0] != 'v' {
+		return ""
+	}
+	return version[1:]
+}
+
+func checkUpdate() (bool, error) {
+	latest, found, err := selfupdate.DetectLatest(updateEndpoint)
+	if err != nil {
+		return false, err // unable to update due to GH error
+	}
+
+	parsedVersion := getSemver()
+	updateLogger.Debugf("checking version for updates (%s -> %s)", version, parsedVersion)
+	v, err := semver.Parse(parsedVersion)
+	if err != nil {
+		return false, fmt.Errorf("invalid version; you may be using a development binary")
+	}
+	if !found || latest.Version.Equals(v) {
+		return false, nil
+	}
+	return true, nil
+}
+
+func doSelfUpdate() error {
+	parsedVersion := getSemver()
+	v, err := semver.Parse(parsedVersion)
+	if err != nil {
+		return errors.New("invalid version; you may be using a development binary")
+	}
+	latest, err := selfupdate.UpdateSelf(v, updateEndpoint)
+	if err != nil {
+		return err
```

Comment:
Use `fmt.Errorf` to annotate this error. It helps us trace errors through the stack.